server/datastore/mysql/mysql.go

```diff
@@ -312,13 +357,28 @@ func (d *Datastore) Drop() error {
 
 // HealthCheck returns an error if the MySQL backend is not healthy.
 func (d *Datastore) HealthCheck() error {
-	_, err := d.db.Exec("select 1")
-	return err
+	if _, err := d.writer.Exec("select 1"); err != nil {
+		return err
+	}
+	if d.readReplicaConfig != nil {
+		var dst int
+		if err := d.reader.Get(&dst, "select 1"); err != nil {
```

Comment:
I'm assuming we want to hit the replica too for health checks and raise if it fails, as it might mean reads will all fail.